rotkehlchen/db/updates.py

```diff
@@ -88,49 +90,175 @@ def update_spam_assets(self, local_version: int, to_version: int) -> None:
+            'SELECT value FROM settings WHERE name=?',
+            (update_type.serialize(),),
+        ).fetchone()
+        return int(found_version[0]) if found_version is not None else 0
+
+    @staticmethod
+    def _get_file_content(
+            file_url: str,
+            update_type: UpdateType,
+    ) -> tuple[bool, list[dict[str, Any]]]:
+        """Retrieves the content of data to be updated.
+
+        Returns True/False for success along with the content.
+        """
+        try:
+            updates = query_file(file_url, is_json=True)
+        except RemoteError as e:
+            log.warning(f'Failed to update {update_type.value} due to {str(e)}')
+            return False, []
+
+        update_content = updates.get(update_type.value)
+        if update_content is None:
+            log.error(f'Remote update {file_url} does not contain {update_type.value} key')
+            return False, []
+
+        return True, update_content
+
+    def _update_user_nodes(
+            self,
+            existing_default_nodes: list[tuple[Any, ...]],
+            new_default_nodes: list[tuple[Any, ...]],
+    ) -> None:
+        """Updates the user nodes using the default nodes from the global db.
+
+        This function does the following:
+        1. Compares the previous default rpc nodes to the new ones, the difference is deleted from the user db.
+        2. Adds the new default rpc nodes to the user db.
+
+        indexes 1 & 2 -> endpoint of node
+        indexes 5 & 6 -> blockchain of node
+        """  # noqa: E501
+        # check for nodes to delete for the user
+        nodes_to_delete = (
+            {(node[2], node[6]) for node in existing_default_nodes} -
+            {(node[1], node[5]) for node in new_default_nodes}
+        )
+        if len(nodes_to_delete) != 0:
+            log.debug(f'Deleting {nodes_to_delete} nodes from user database...')
+            with self.user_db.user_write() as write_cursor:
+                write_cursor.executemany(
+                    'DELETE FROM rpc_nodes WHERE endpoint=? AND blockchain=?',
+                    list(nodes_to_delete),
+                )
+
+        with self.user_db.conn.read_ctx() as cursor:
+            user_rpc_nodes = {
+                (entry[2], entry[6])
+                for entry in cursor.execute('SELECT * FROM rpc_nodes')
+            }
+
+        # determine the nodes to add to the user db by
+        # checking if it's not already present in the user db.
+        nodes_to_add = [
+            node_to_add
+            for node_to_add in new_default_nodes
+            if (node_to_add[1], node_to_add[5]) not in user_rpc_nodes
+        ]
+        with self.user_db.user_write() as write_cursor:
+            log.debug(f'Adding {nodes_to_add} nodes to the user database...')
+            write_cursor.executemany(
+                'INSERT OR REPLACE INTO rpc_nodes(name, endpoint, owned, active, weight, blockchain) '  # noqa: E501
```

Comment:
the replace here is useless. Since the primary key is the identifier.
```suggestion
                'INSERT INTO rpc_nodes(name, endpoint, owned, active, weight, blockchain) '  # noqa: E501
```